service/worker/scanner/replication/replication.go

```diff
@@ -174,43 +215,117 @@ func GenReplicationWorkflow(ctx workflow.Context, options ForceReplicationOption
 		}
 	}
 
-	if options.AllowedLaggingSeconds <= 0 {
-		return nil // no need to wait
+	return nil
+}
+
+func NamespaceHandoverWorkflow(ctx workflow.Context, params NamespaceHandoverParams) error {
+	// validate input params
+	if len(params.Namespace) == 0 {
+		return errors.New("InvalidArgument: Namespace is required")
 	}
-	// ** Step 3, get replication status **
-	ao3 := workflow.ActivityOptions{
-		StartToCloseTimeout: time.Second * 30,
+	if len(params.RemoteCluster) == 0 {
+		return errors.New("InvalidArgument: RemoteCluster is required")
+	}
+	if params.AllowedLaggingSeconds <= minimumAllowedLaggingSeconds {
+		params.AllowedLaggingSeconds = minimumAllowedLaggingSeconds
+	}
+	if params.HandoverTimeoutSeconds <= minimumHandoverTimeoutSeconds {
+		params.HandoverTimeoutSeconds = minimumHandoverTimeoutSeconds
+	}
+
+	retryPolicy := &temporal.RetryPolicy{
+		InitialInterval:    time.Second,
+		MaximumInterval:    time.Second,
+		BackoffCoefficient: 1,
+	}
+	ao := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Second * 10,
 		RetryPolicy:         retryPolicy,
 	}
-	ctx3 := workflow.WithActivityOptions(ctx, ao3)
-	var repStatus replicationStatus
-	err = workflow.ExecuteActivity(ctx3, a.GetMaxReplicationTaskIDs).Get(ctx3, &repStatus)
+	ctx = workflow.WithActivityOptions(ctx, ao)
+
+	var a *activities
```

Comment:
a?

Reply:
That is what we suggested it in all our samples. :)